### PR TITLE
add notification for linux issue

### PR DESCRIPTION
**Pull request by sbatten**
Notification of workaround for #62593

Goals:
1) Recommend user to switch to custom title bar on Linux
2) Only show if they use the native title bar and the menu bar is enabled (!hidden).
3) Link to issue describing the problem.

### CODE REVIEW

File: src/vs/workbench/browser/parts/titlebar/menubarControl.ts [hunk 1]
```diff
@@ -436,6 +443,43 @@ export class MenubarControl extends Disposable {
...
+			return;
+		}
+
+		if (!this.storageService.getBoolean('menubar/electronFixRecommended', StorageScope.GLOBAL, false)) {
+			if (this.currentMenubarVisibility === 'hidden' || this.currentTitlebarStyleSetting === 'custom') {
+				// Issue will not arise for user, abort notification
+				return;
+			}
+
+			const message = nls.localize('menubar.electronFixRecommendation', 'If you experience a low-contrast issue with the menu bar, we recommend trying out the custom title bar.');
```
Comment (roblourens) [inline src/vs/workbench/browser/parts/titlebar/menubarControl.ts]: "low-contrast issue" be more specific? I wouldn't know what this means

File: src/vs/workbench/browser/parts/titlebar/menubarControl.ts [hunk 2]
```diff
@@ -436,6 +443,43 @@ export class MenubarControl extends Disposable {
...
+			this.notificationService.prompt(Severity.Info, message, [
+				{
+					label: nls.localize('tryIt', 'Try it'),
```
Comment (roblourens) [inline src/vs/workbench/browser/parts/titlebar/menubarControl.ts]: How about just opening the settings editor? "Try it" implies that I can try this out with an easy way to get back but here the user won't know how to undo this.

File: src/vs/workbench/browser/parts/titlebar/menubarControl.ts [hunk 3]
```diff
@@ -436,6 +443,43 @@ export class MenubarControl extends Disposable {
...
+					run: () => {
+						// Don't recommend this again
+						this.storageService.store('menubar/electronFixRecommended', true, StorageScope.GLOBAL);
+						return this.configurationService.updateValue('window.titleBarStyle', 'custom', ConfigurationTarget.USER);
+					}
+				},
+				{
+					label: nls.localize('moreInfo', 'More info'),
+					run: () => {
+						window.open('https://github.com/Microsoft/vscode/issues/62593');
```
Comment (roblourens) [inline src/vs/workbench/browser/parts/titlebar/menubarControl.ts]: aka.ms url?

File: src/vs/workbench/browser/parts/titlebar/menubarControl.ts [hunk 4]
```diff
@@ -436,6 +443,43 @@ export class MenubarControl extends Disposable {
...
+					}
+				},
+				{
+					label: nls.localize('neverShowAgain', 'Don\'t Show Again'),
```
Comment (roblourens) [inline src/vs/workbench/browser/parts/titlebar/menubarControl.ts]: Double quotes on the localized strings

**Comment by roblourens** [comment]
Could this be scoped to only be shown to people using 18.10?

File: src/vs/workbench/browser/parts/titlebar/menubarControl.ts (hunk 2 above)
Comment (roblourens) [reply src/vs/workbench/browser/parts/titlebar/menubarControl.ts]: Or change the string to something that sounds more permanent

**Comment by sbatten** [comment]
@roblourens while the issue was found on 18.10, other distros with similar themes, would encounter the issue as well.

File: src/vs/workbench/browser/parts/titlebar/menubarControl.ts [hunk 5]
```diff
@@ -436,6 +445,41 @@ export class MenubarControl extends Disposable {
...
+			const message = nls.localize('menubar.electronFixRecommendation', "If you experience hard to read text in the menu bar, we recommend trying out the custom title bar.");
+			this.notificationService.prompt(Severity.Info, message, [
+				{
+					label: nls.localize('tryIt', "Try it"),
+					run: () => {
+						return this.preferencesService.openGlobalSettings(undefined, { query: 'window.titleBarStyle' });
+					}
+				},
+				{
+					label: nls.localize('moreInfo', "More info"),
```
Comment (roblourens) [inline src/vs/workbench/browser/parts/titlebar/menubarControl.ts]: "Don't Show Again" capitalizes every word, the others should match

**Review by roblourens — APPROVED** [review]

**Pull request merged**